### ISSUE TITLE
[Snap] nteract not detecting python kernel when installed via Snap.

### ISSUE THREAD

**Issue by ramantehlan** [nteract/archived-desktop-app#5054]
When nteract is installed via [snap](https://snapcraft.io/nteract), it is not detecting the python kernel. 

Not sure what is causing the issue. From what I understand about the snap working, It stores its binary in the `/snap` folder, with no dependencies on the system libraries, but when you install nteract with `.deb`, it stores its binary in the `/bin` folder and does depend on system libraries present in the `/lib` folder.

**Comment by jarmokivekas** [comment]
I was setting up my nteract devel environment, installing node from snap, and got this message:
```
> snap install --edge node 
error: This revision of snap "node" was published using classic confinement and thus may perform
       arbitrary system changes outside of the security sandbox that snaps are usually confined to,
       which may put your system at risk.

       If you understand and want to proceed repeat the command including --classic.
```

Might be that this "classic confinement" is what nteract needs. Python notebooks and using the python kernel definitely seem to fit the description in the warning message. Currently when installing nteract, snap does not ask for `--classic`, which means it is probably under some more restrictive confinement scheme. (Speculation, I'm no snap guru)

**Comment by BenRussert** [comment]
@jarmokivekas I think you are correct about confinement being the issue here. 

According to [the snap docs](https://snapcraft.io/docs/snap-confinement) you can't access any system resources (in nteract's case e.g. files, node, etc) without classic confinement or [interfaces](https://snapcraft.io/docs/interface-management). 

It looks the "classic" option is likely what we need, but requires some kind of [manual approval process](https://forum.snapcraft.io/t/process-for-reviewing-classic-confinement-snaps/1460), not sure the difficulty of approval there, but this is what atom, vscode do probably for similar reasons that nteract would.

**Comment by ramantehlan** [comment]
I have submitted a "classic confinement" request for nteract [here](https://forum.snapcraft.io/t/classic-confinement-request-nteract-desktop-application/16697). 

The process states the publisher should submit a request, which now I am not. I have mentioned that in the request, and if that will be a problem, we can request the same from the official account.

**Comment by captainsafia** [comment]
I'll keep any eye out for something from the Snap team around this.

**Comment by ramantehlan** [comment]
Right now the nteract desktop is on the edge channel on the snap store. However, classic confinement is only supported for the stable channel.

If the request is approved, what do we do in that stage? Do we change the channel to stable or take the snap off the store and put it back once it is stable?

**Comment by captainsafia** [comment]
We can move to the stable channel if that will unblock labelling it with classic confinement.

**Comment by ramantehlan** [comment]
Yes, I think we should move it to the sable channel to be able to get classic confinement.

**Comment by ramantehlan** [comment]
We have to the green flag from the snapcraft team and now nteract snap has the `classic confinement`. 

However, we will have to move the snap to the stable channel. Also, we should add a note to the description asking users who have already installed the snap to upgrade it with classic confinement using below command.

```
sudo snap refresh nteract --classic
```

**Comment by jarrodu** [comment]
I am also running into this issue with the Julia kernel. Is there any update on when nteract will be available on the stable channel?

**Comment by ramantehlan** [comment]
@captainsafia I don't have access to the snap package anymore. If we can move it to stable channel, it should work.
